scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateIfTaskHandler.java

```diff
@@ -0,0 +1,190 @@
+			}
+		}
+
+		// even though the targetElse is not going to be executed, a
+		// dependency on initiator still makes sense and would help
+		// reconstruct the job graph on the client
+		targetElse.addDependence(initiator);
+		changesInfo.taskUpdated(targetElse);
+
+		for (InternalTask it : elseTasks) {
+			it.setFinishedTime(descriptor.getFinishedTime() + 1);
+			it.setStatus(TaskStatus.SKIPPED);
+			it.setExecutionDuration(0);
+
+			changesInfo.taskSkipped(it);
+			internalJob.setNumberOfPendingTasks(internalJob.getNumberOfPendingTasks() - 1);
+			internalJob.setNumberOfFinishedTasks(internalJob.getNumberOfFinishedTasks() + 1);
+			logger.info("Task " + it.getId() + " will not be executed");
+		}
+
+		// plug the branch in the descriptor
+		TaskId joinId = null;
+		if (targetJoin != null) {
+			joinId = targetJoin.getId();
+		}
+		internalJob.getJobDescriptor().doIf(initiator.getId(), branchStart.getId(), branchEnd.getId(), joinId,
+				targetElse.getId(), elseTasks);
+
+		((JobInfoImpl) internalJob.getJobInfo()).setTasksChanges(changesInfo, internalJob);
+		// notify frontend that tasks were modified
+		if (frontend != null) {
+			frontend.jobStateUpdated(internalJob.getOwner(),
+					new NotificationData<>(SchedulerEvent.TASK_SKIPPED, internalJob.getJobInfo()));
+		}
+		((JobInfoImpl) internalJob.getJobInfo()).clearTasksChanges();
+
+		// no jump is performed ; now that the tasks have been plugged
+		// the flow can continue its normal operation
+		internalJob.getJobDescriptor().terminate(taskId);
+
+		return true;
+	}
+
+	private InternalTask[] searchIfElseJoinTasks(FlowAction action, InternalTask initiator) {
+		InternalTask targetIf = null;
+		InternalTask targetElse = null;
+		InternalTask targetJoin = null;
+
+		// search for the targets as perfect matches of the unique name
+		for (InternalTask it : internalJob.getIHMTasks().values()) {
+
+			// target is finished : probably looped
+			if (it.getStatus().equals(TaskStatus.FINISHED) || it.getStatus().equals(TaskStatus.SKIPPED)) {
+				continue;
+			}
+			if (action.getTarget().equals(it.getName())) {
+				if (it.getIfBranch().equals(initiator)) {
+					targetIf = it;
+				}
+			} else if (action.getTargetElse().equals(it.getName())) {
+				if (it.getIfBranch().equals(initiator)) {
+					targetElse = it;
+				}
+			} else if (action.getTargetContinuation().equals(it.getName())) {
+				InternalTask up = internalJob.findTaskUp(initiator.getName(), it);
+				if (up != null && up.equals(initiator)) {
+					targetJoin = it;
+				}
+			}
+
+		}
+
+		boolean searchIf = (targetIf == null);
+		boolean searchElse = (targetElse == null);
+		boolean searchJoin = (targetJoin == null);
+
+		// search of a runnable perfect match for the targets failed;
+		// the natural target was iterated, need to find the next iteration
+		// which is the the one with the same dup index and base name,
+		// but the highest iteration index
+		for (InternalTask it : internalJob.getIHMTasks().values()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS135)